src/LndClient/Data/Invoice.hs

```diff
@@ -25,22 +25,25 @@ data Invoice
   = Invoice
       { rHash :: RHash,
         amtPaidSat :: Maybe Text,
-        creationDate :: Text,
+        creationDate :: Maybe Text,
         settleDate :: Maybe Text,
         value :: MoneyAmount,
-        expiry :: Text,
+        expiry :: Maybe Text,
         settled :: Maybe Bool,
         settleIndex :: Maybe Text,
         descriptionHash :: Maybe Text,
         memo :: Maybe Text,
         paymentRequest :: Maybe Text,
         fallbackAddr :: Maybe Text,
-        cltvExpiry :: Text,
+        cltvExpiry :: Maybe Text,
         private :: Maybe Bool,
-        addIndex :: Maybe Text,
+        addIndex :: Text,
```

Comment:
I don't think invoice type which is accepted by `addInvoice` RPC should be the same like invoices which are returned back from LND by other methods. Because for second case there are probably some fields which should be required (like "expiry", "creationDate" etc) which are probably guaranteed by LND (just guessing)

And about "addIndex" previously it has been auto-generated by LND, and now library user should generate it?

Reply:
Agree, fixed, using separate type for adding invoice.